cli/pkg/validate/required_keys_test.go

```diff
@@ -356,3 +357,189 @@ func Test_IsNumber(t *testing.T) {
+		},
+		{
+			name: "property name with both leading and trailing whitespace",
+			properties: map[catalog.EntityGroup][]catalog.Property{
+				"test-group": {
+					{
+						LocalID:     "both-space-prop",
+						Name:        " Property With Both Spaces ",
+						Description: "Property with both leading and trailing whitespace",
+						Type:        "string",
+					},
+				},
+			},
+			expectedErrors: 1,
+			errorContains:  "property name cannot have leading or trailing whitespace characters",
+		},
+		{
+			name: "property name with tab characters",
+			properties: map[catalog.EntityGroup][]catalog.Property{
+				"test-group": {
+					{
+						LocalID:     "tab-prop",
+						Name:        "\tProperty With Tab\t",
+						Description: "Property with tab characters",
+						Type:        "string",
+					},
+				},
+			},
+			expectedErrors: 1,
+			errorContains:  "property name cannot have leading or trailing whitespace characters",
+		},
+		{
+			name: "property name with newline characters",
+			properties: map[catalog.EntityGroup][]catalog.Property{
+				"test-group": {
+					{
+						LocalID:     "newline-prop",
+						Name:        "\nProperty With Newline\n",
+						Description: "Property with newline characters",
+						Type:        "string",
+					},
+				},
+			},
+			expectedErrors: 1,
+			errorContains:  "property name cannot have leading or trailing whitespace characters",
+		},
+		{
+			name: "property name with internal spaces (should be valid)",
+			properties: map[catalog.EntityGroup][]catalog.Property{
+				"test-group": {
+					{
+						LocalID:     "internal-space-prop",
+						Name:        "Property With Internal Spaces",
+						Description: "Property with internal spaces which should be allowed",
+						Type:        "string",
+					},
+				},
+			},
+			expectedErrors: 0,
+		},
+		{
+			name: "empty property name should trigger mandatory field error, not whitespace error",
+			properties: map[catalog.EntityGroup][]catalog.Property{
+				"test-group": {
+					{
+						LocalID:     "empty-name-prop",
+						Name:        "",
+						Description: "Property with empty name",
+						Type:        "string",
+					},
+				},
+			},
+			expectedErrors: 1,
+			errorContains:  "id, name and type fields on property are mandatory",
+		},
+		{
+			name: "multiple properties with different whitespace issues",
+			properties: map[catalog.EntityGroup][]catalog.Property{
+				"test-group": {
+					{
+						LocalID:     "leading-space-prop",
+						Name:        " Leading Space",
+						Description: "Property with leading space",
+						Type:        "string",
+					},
+					{
+						LocalID:     "trailing-space-prop",
+						Name:        "Trailing Space ",
+						Description: "Property with trailing space",
+						Type:        "string",
+					},
+					{
+						LocalID:     "valid-prop",
+						Name:        "Valid Property",
+						Description: "Valid property",
+						Type:        "string",
+					},
+				},
```

Comment:
I would rather have this single test case and remove all the others. They add more noise, whatever we are testing is contained within this set of inputs.